zulipterminal/model.py

```diff
@@ -945,24 +947,44 @@ def get_stream_by_id(streams: List[StreamData], stream_id: int
                         for user_id in user_ids:
                             subscribers.remove(user_id)
 
+    @asynch
+    def _show_typing_animation(self, recipient: Dict[str, Any]) -> None:
+        start_time = datetime.now()
+        # Restore the footer if there hasn't been a corresponding typing
+        # 'stop' event after a 'start' event for greater than 15 seconds.
+        max_duration = timedelta(seconds=TYPING_STARTED_EXPIRY_PERIOD)
+
+        for dots in itertools.cycle(['', '.', '..', '...']):
```

Comment:
Could we use the `next` function with a `lambda` to iterate over the dots? That'd be a bit cleaner.

Reply:
I don't quite get this, next can be used only when a generator object is created right? Is lambda used to create this object? That would leads to more lines of code for getting the same functionality?

Reply:
Agreed; it would require more lines. However, it would facilitate is the separation of concerns. @neiljp Thoughts on this?

Reply:
@preetmishra I'd prefer to see the comparison before judging.

Reply:
@preetmishra I've re-implemented this part with minor changes. I've not used lambda as I was getting an error but I think this is pretty much what you had in mind. It looks much more better and readable now though its the same number of lines because of the if statement coming out of the loop.